treewide: remove `multiArch=false` and wrong usage of `extraPkgs`

SuperSamus opened the pull request:
## Description of changes

Cleanup after the work of #240860.
Motivation: prevent new packages to copy from outdated examples.
Didn't test.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/3593

SuperSandro2000 commented:
fyi @Atemu

Atemu commented:
This appears to change eval on many of the touched packages, what's the difference?

SuperSamus commented:
> This appears to change eval on many of the touched packages, what's the difference?

Sorry for the late response, but I don't know.

Atemu commented:
Please investigate that, I do not feel comfortable merging this otherwise.

SuperSamus commented:
Alright, I found the cause.
Before, some packages were listed twice:
- In `multiPkgs` (because they were there by default)
- In `extraPkgs` (because of `extraPkgs = pkgs: (appimageTools.defaultFhsEnvArgs.multiPkgs pkgs)`)

By removing the latter, these packages are listed only once.

Example with Zettlr (I took the pkgs from the derivations and prettified the JSON).
- After PR: 
[h3py958xa6rmhv10q9xizlx5y5qlhpqf-zettlr-3.0.2-usr-target.drv.json](https://github.com/NixOS/nixpkgs/files/15094124/h3py958xa6rmhv10q9xizlx5y5qlhpqf-zettlr-3.0.2-usr-target.drv.json)
- Before PR: 
[xjl0fnnwy3ad1czrnk6slci2kf4brp7n-zettlr-3.0.2-usr-target.drv.json](https://github.com/NixOS/nixpkgs/files/15094152/xjl0fnnwy3ad1czrnk6slci2kf4brp7n-zettlr-3.0.2-usr-target.drv.json)

By using a text comparison tool (or a [duplicate counter](https://www.somacon.com/p568.php)), you can see that "After PR" only removes duplicates.

I only tested Zettlr, but I suppose it's the same for the others.

Atemu commented:
Could you squash this into one commit? It's a tree-wide change, so that's one logical step.

pbsds commented:
Result of `nixpkgs-review pr 279260` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>13 packages built:</summary>
  <ul>
    <li>Sylk</li>
    <li>bloomrpc</li>
    <li>chrysalis</li>
    <li>flexoptix-app</li>
    <li>joplin-desktop</li>
    <li>marktext</li>
    <li>mycrypto</li>
    <li>notable</li>
    <li>plexamp</li>
    <li>polypane</li>
    <li>station</li>
    <li>tusk</li>
    <li>zettlr</li>
  </ul>
</details>

Atemu commented:
There's a conflict now. Could you also add a commit message that explains the background of the change? The contributor's manual has a section on that. It should also say something to the effect of `treewide: remove multiArch=false from buildFHSEnv usages`.

Other than that, this LGTM.

SuperSamus commented:
So, I just noticed that there are other packages that do `extraPkgs = pkgs: appimageTools.defaultFhsEnvArgs.multiPkgs pkgs;`, which I didn't notice before because they didn't set `multiArch = false;`.
I guess I should make a separate commit for that.

SuperSamus committed:
treewide: remove `multiArch=false` from `buildFHSEnv` usages
Cleanup after the work of #240860.
Also preventing new packages to copy from outdated ones.

SuperSamus commented:
35 packages that either do `extraPkgs = pkgs: appimageTools.defaultFhsEnvArgs.multiPkgs pkgs;` or accidentally use the top-level input.
With the latter I mean:
```nix
{ appimageTools, pandoc }:
appimageTools.wrapType2 {
  # Haa, the joys of being bamboozled by the `with;` keyword!
  # I can say that I've had experience with that, I'm definitively still not angry about the time I wasted for #248898.
  extraPkgs = pkgs: with pkgs; [ pandoc ];
}
```
*Maybe* fixing this should be a separate PR.

SuperSamus commented:
So, there are 5 packages that use `wrapAppImage` instead of `wrapType2`.
Documentation doesn't mention the existence of the former at all.
Also, while in `wrapType2` `extraPkgs` defaults to `[ ]`, the same doesn't apply to `wrapAppImage` (which has no default).
It _seems_ to me that  `wrapAppImage` is only meant to be called by `wrapType2` and nothing else.

Should I convert all the usages of `wrapAppImage` to `wrapType2`?
Or just put `extraPkgs ? pkgs: [ ]` in `wrapAppImage`?

pbsds commented:
`extraPkgs ? pkgs: [ ]` strikes me as a sane default

Atemu reviewed (approved):
I think we're looking at remnants of previous iterations of this interface where there was support for multiple versions of the appimage format. These days, everything just uses wrapType2 IIRC.

Appimage support in general needs some cleanup.

This LGTM.

Atemu commented on pkgs/build-support/appimage/default.nix:
```diff
@@ -40,7 +40,7 @@ rec {
   wrapAppImage = args@{
     src,
-    extraPkgs,
+    extraPkgs ? pkgs: [],
```
Could you split this into a separate commit? It doesn't belong with the others.

SuperSamus committed:
wrapAppImage: default `extraPkgs` to `pkgs: [ ]`
Sane default in preparation for the next commit.

Atemu commented on pkgs/by-name/pc/pcsx2/GameIndex.yaml:
?

SuperSamus replied:
Unrelated thing slipped in, sorry.

SuperSamus committed:
treewide: fix use of `extraPkgs` in AppImages
Mostly removes unnecessary use of `extraPkgs = pkgs: appimageTools.defaultFhsEnvArgs.multiPkgs pkgs;`
This caused some packages to be listed twice.

Also, fix some styling, and accidental use of top-level packages (sometimes due to the `with;` keyword, e.g. on `beeper`).
Remove inclusions of `bash`, since `bashInteractive` is already present by default.

Atemu commented:
Thank you!
